src/JWT/Algorithms/HMACSHAAlgorithmFactory.cs

```diff
@@ -25,7 +25,7 @@ public virtual IJwtAlgorithm Create(JwtHashAlgorithm algorithm)
                 case JwtHashAlgorithm.RS256:
                     throw new NotSupportedException($"For algorithm {nameof(JwtHashAlgorithm.RS256)} please create custom factory by implementing {nameof(IAlgorithmFactory)}");
                 default:
-                    throw new InvalidOperationException($"Algorithm {algorithm} is not supported.");
+                    throw new NotSupportedException($"Algorithm {algorithm} is not supported.");

```

Comment:
Technically a breaking change. Will this get a version bump?

Reply:
Yes, this and few minor more. So wondering too should we bump to 4.0 or keep 3.2